processor/schematransformerprocessor/internal/schema/indentifer_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema_test
```

Comment:
Sorry, do you mind giving an example?

Reply:
I might certainly be wrong, but I just picked a random component and it's indeed using the same package for both the tests and the test subject:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b6a9b30160f6e82d61b95f11568be8f415a8c93d/processor/cumulativetodeltaprocessor/factory_test.go#L15

Reply:
Yes, we typically test from the same package to ensure that unexported types and functions can be tested in isolation where needed.  Occasionally it may be useful to test from an external package perspective, but that should generally be reserved for cases such as validating that something is/isn't possible from outside the package under test.

Reply:
Right! Sorry, Brain failed to engage :) 

Yeah, that is completely fine. I typically prefer doing it this way so that changes to functions or signatures are easily caught and super obvious. 

Happy to move the package to use the exported package instead of the test package to make sure all code is covered.